samples/NativeLibrary/Class1.cs

```diff
@@ -51,5 +52,39 @@ public static IntPtr sumstring(IntPtr first, IntPtr second)
             // Return pointer
             return sumPointer;
         }
+
+        [UnmanagedCallersOnly(EntryPoint = "mergestrings")]
+        public static void MergeStrings(IntPtr oput, IntPtr strArr, int count)
+        {
+            string mainString = String.Empty;
+            for (int i = 0; i < count; i++)
+            {
+                int currentOffset = i * IntPtr.Size;
+                IntPtr curPtr = Marshal.ReadIntPtr(strArr, currentOffset);
+                string curString = Marshal.PtrToStringAnsi(curPtr);
+                mainString += $"String number {i} : {curString} \n";
+            }
+
+            // Let's 'send back' our string to the C program
+            // We need to convert it to an array of ASCII representations of our characters
+            // After that,we'll copy the bytes array to the 'oput' memory location
+            byte[] bytesFinalString = Encoding.ASCII.GetBytes(mainString);
+            Marshal.Copy(bytesFinalString, 0, oput, mainString.Length);
+        }
+
+        struct myStruct
+        {
+            public IntPtr name;
+            public int value;
+        }
+
+        [UnmanagedCallersOnly(EntryPoint = "popstruct")]
+        public static void PopulateStruct(IntPtr structurePtr)
+        {
+            myStruct structure = Marshal.PtrToStructure<myStruct>(structurePtr);
```

Comment:
For blittable struct, it is more appropriate to just use C# pointers. 

```csharp
public static void PopulateStruct(myStruct* structurePtr)
{
    structurePtr->name = Marshal.StringToHGlobalAnsi(Guid.NewGuid().ToString());
    structurePtr->value = new Random().Next();
}
```

Reply:
Definitely better! I just put the method in unsafe context so i can use pointers